app/init/app.ts

```diff
@@ -55,25 +51,19 @@ export async function initialize() {
 }
 
 export async function start() {
-    if (baseAppInitialized) {
-        // Clean relevant information on ephemeral stores
-        NavigationStore.reset();
-        EphemeralStore.setCurrentThreadId('');
-        EphemeralStore.setProcessingNotification('');
-    }
+    // Clean relevant information on ephemeral stores
+    NavigationStore.reset();
+    EphemeralStore.setCurrentThreadId('');
+    EphemeralStore.setProcessingNotification('');
 
     await initialize();
 
-    if (!mainAppInitialized) {
-        mainAppInitialized = true;
+    PushNotifications.init(serverCredentials.length > 0);
```

Comment:
Why do we want to register this when the share extension and not the app has been opened? Same question for the WebSockets, screens and navigation? I really don't think this is the right solution, not saying that what we have is, but we need to identify at least on Android if the `Navigation.events().registerAppLaunchedListener` was triggered by the extension or the main app.. maybe you can check here for Android if the activity is the main activity by `import {isMainActivity} from '@utils/helpers';`

Reply:
If I am understanding the code correctly (I think I did some test around that) the share extension never calls this piece of code. The share extension call the `initialize` function on `share_extension/index.tsx` not the `start` function. The `start` function is only called by the main activity (if I am not mistaken).

So... since we are only changing the behavior of the `start` function, the previous logic regarding the share extension should have not changed.